Joedmin-hw3/Joedmin-hw3/Program.cs

```diff
@@ -0,0 +1,153 @@
+                        using (var stream = new StreamWriter("Users.txt", true))
+                        {
+                            while (true)
+                            {
+                                try
+                                {
+                                    string[] credentials = GetCredencials();
+
+                                    SaveToDictionary(usersAndPasswords, credentials);
+                                    stream.WriteLine($"{credentials[0]} {credentials[1]}");
+                                    Console.WriteLine("Hello!");
+                                    break;
+                                }
+                                catch (DuplicateUserCredentialsException ex)
+                                {
+                                    Console.WriteLine($"{ex.Message}\nPlease try again.");
+                                }
+                            }
+                        }
+                    }
+                    catch (FileNotFoundException)
+                    {
+                        throw new UsersNotFoundException("File Users.txt was not found. Make sure it is in the same directory as the .exe file.");
+                    }
+                }
+            }
+            catch (UsersNotFoundException ex)
+            {
+                Console.WriteLine(ex.Message);
+            }
+            Console.ReadKey(true);
+        }
+
+        // Get user's user name and password
+        static string[] GetCredencials()
+        {
+            Console.Write("Enter your username: ");
+            string username = Console.ReadLine();
+
+            Console.Write("Enter your password: ");
+            string password = Console.ReadLine();
+
+            return new string[] { username.Replace(" ", String.Empty), password.Replace(" ", String.Empty) };
+        }
+
+        // Reading users.txt and saving it's whole content to a dictionary
+        static Dictionary<string, string> GetUsersAndPasswords(string path)
+        {
+            try
+            {
+                using (var stream = new StreamReader(path))
+                {
+                    string[] usersAndPasswords = stream.ReadToEnd().Split(Environment.NewLine);
+
+                    Dictionary<string, string> usersInfos = new Dictionary<string, string>();
+
+                    foreach (var item in usersAndPasswords)
+                    {
+                        string[] line = item.Split(' ');
+                        SaveToDictionary(usersInfos, line);
+                    }
+                    return usersInfos;
+                }
+            }
+            catch (FileNotFoundException)
+            {
+                throw new UsersNotFoundException("File Users.txt was not found. Make sure it is in the same directory as the .exe file.");
+            }
+
+        }
+
+        // Adding user's data from 1 line of a file to a dictionary with check of duplicated user names (keys)
```

Comment:
Use XML docs instead of normal comments for documentation.
`///` and it will generate an annotation. This will enable some intellisense features- like hovering over the method and seeing what the comment is.